schemas/applications/schema/joint_positions.schema.json

```diff
@@ -0,0 +1,36 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Joint Positions",
+  "description": "A description of the joint configurations available in the application",
+  "type": "object",
+  "additionalProperties": false,
+  "patternProperties": {
+    "^[a-z]([a-z0-9_]?[a-z0-9])*$": {
+      "title": "Joint Positions",
+      "description": "The names and positions of a static joint configuration in the application",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "names": {
+          "title": "Joint Names",
+          "description": "The names of the joints in the same order as the positions",
+          "type": "array",
+          "minItems": 1,
+          "uniqueItems": true,
+          "items": "string"
+        },
+        "positions": {
+          "title": "Joint Positions",
+          "description": "The positions of the joints in the same order as the names",
+          "type": "array",
+          "minItems": 1,
+          "items": "number"
+        }
+      },
+      "required": [
```

Comment:
@eeberhard In the parent issue you've mentioned that you can assert equal length, I couldn't find any hints that this is actually possible. Do you know how?

Reply:
you're right, I don't think it's easily possible. You can perform validation on the number of items in an array, and you can also use dependent schemas / conditional sub-schemas to say "if this property exists, this other condition must be met also". You can therefore say something along the lines of "if joint names exists and has 4 items, then positions must also exist with exactly 4 items", but using the length of the first property to validate the second property does not work in a general case as far as I can tell.

In any case there is also no cross-validation possible for the order of the items, and ultimately that is the agreed compromise of this representation